pkg/server/cluster_server.go

```diff
@@ -129,6 +140,17 @@ func (cs *clusterServer) GetConfig(cr poolRequest) (*runtime.RawExtension, error
 		return nil, fmt.Errorf("could not get controllerconfig: %w", err)
 	}
 
+	// we cannot mock this in a test env
```

Comment:
note: You can actually :smile:

It requires a bit of setup with `FakeClient` and whatnot.

Reply:
i tried, but the functions required to fulfill the interfaces needs are impossible (i think?) to make fake equivalents for. There is one extra functions that the other types don't have that returns a lister you can't make manually.